ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/StreamStreamJoinBuilder.java

```diff
@@ -85,8 +85,16 @@ private StreamStreamJoinBuilder() {
     final JoinParams joinParams = JoinParamsFactory
         .create(join.getKeyColName(), leftSchema, rightSchema);
 
-    final JoinWindows joinWindows =
+    JoinWindows joinWindows =
         JoinWindows.of(join.getBeforeMillis()).after(join.getAfterMillis());
+
+    // Grace, as optional, helps to identify if a user use the new GRACE PERIOD syntax in the
+    // join window or not. If used, then we'll call the new KStreams API ofSizeAndGrace() which
+    // enables the bugfix with left/outer joins (see KAFKA-10847).
+    if (join.getGraceMillis().isPresent()) {
+      joinWindows = joinWindows.grace(join.getGraceMillis().get());
```

Comment:
This is calling the old/exiting `grace()` and thus it does not enable this fix. I think we need to change this (starting in L88):
```
final JoinWindows joinWindows;
if (join.getGraceMillis().isPresent()) {
    joinWindows = JoinsWindow.ofTimeDifferenceAndGrace(
        join.getBeforeMillis(),
        join.getGraceMillis().get()
    ).after(join.getAfterMillis();
} else {
    joinWindows  = JoinWindows.of(join.getBeforeMillis()).after(join.getAfterMillis())
}
```

Reply:
I think #7695 would update this.

Reply:
Right. I need https://github.com/confluentinc/ksql/pull/7695 merged before I make the changes for the new `ofTimeDifferenceAndGrace`.